filament/backend/src/metal/MetalDriver.mm

```diff
@@ -1267,29 +1267,29 @@
     // Enumerate all the sampler buffers for the program and check which textures and samplers need
     // to be bound.
 
-    id<MTLTexture> texturesToBind[SAMPLER_BINDING_COUNT] = {};
-    id<MTLSamplerState> samplersToBind[SAMPLER_BINDING_COUNT] = {};
+    id<MTLTexture> texturesToBindVertexShader[backend::MAX_VERTEX_SAMPLER_COUNT] = {};
+    id<MTLSamplerState> samplersToBindVertexShader[backend::MAX_VERTEX_SAMPLER_COUNT] = {};
 
-    enumerateSamplerGroups(program, [this, &texturesToBind, &samplersToBind](
-            const SamplerGroup::Sampler* sampler,
-            uint8_t binding) {
+    enumerateSamplerGroups(program, ShaderType::VERTEX,
```

Comment:
This lambda is nearly identical to one below for fragment textures. Can we re-use it or make it a more generic function?

Reply:
I have tried to remove duplicated code as possible as I can. It's very hard because lambda function needs captured variable and template lambda function is supported since c++20. If you have a better idea please let me know.